flow.py

```diff
@@ -57,17 +62,70 @@ def dest(self):
     def dest(self, dest):
         raise AttributError("Cannot modify a flow's destination node.")
 
-    def make_packet(self, id, src, dest, data):
+    def check_last(self):
+        """Method that checks if all packets have been acknowledged and
+        updating the NetworkSimulator object accordingly
+        """
+        if self.last_acknowledged == self.num_packets:
+            self.ns.num_active_flows -= 1
+
+    def update_flow(a_packet):
+        """Upon receiving an acknowledgement packet, updating the flow's
+        attributes
+
+        Args:
+            a_packet (AcknowledgementPacket): Packet being sent
+                back from host
+        """
+        print("Flow: updated with acknowledgement packet")
+        self.last_acknowledged += 1
+        self.check_last()
+        self.unacknowledged_packets.remove(a_packet)
```

Comment:
isn't the key of self.unacknowledged_packets the id of the packet, not the packet itself?